staticd/static_routes.c

```diff
@@ -302,22 +302,8 @@ struct static_nexthop *static_add_nexthop(struct static_path *pn,
 	default:
 		break;
 	}
-	/*
-	 * Add new static route information to the tree with sort by
-	 * gateway address.
-	 */
-	frr_each(static_nexthop_list, &pn->nexthop_list, cp) {
-		if (nh->type == STATIC_IPV4_GATEWAY
-		    && cp->type == STATIC_IPV4_GATEWAY) {
-			if (ntohl(nh->addr.ipv4.s_addr)
-			    < ntohl(cp->addr.ipv4.s_addr))
-				break;
-			if (ntohl(nh->addr.ipv4.s_addr)
-			    > ntohl(cp->addr.ipv4.s_addr))
-				continue;
-		}
-	}
-	static_nexthop_list_add_after(&(pn->nexthop_list), cp, nh);
```

Comment:
Have a concern that downstream code (in zebra?) may count on this sorting for having deterministic ECMP NH selection.. it probably shouldn't but let's check.

Reply:
Removed this change.